### PR TITLE
[IMP] l10n_es_edi_facturae: add reference tag on invoice lines

### PR DESCRIPTION
**Pull request by alialfie**
When an invoice is linked to a sale order or more, the
ReceiverTransactionReference tag should be included in the Factura-E XML per invoice line with the value of the Customer Reference field of the Sale Order linked to that line. This commit handles this.

task-4134422

### CODE REVIEW

File: addons/l10n_es_edi_facturae/models/account_move.py [hunk 1]
```diff
@@ -159,6 +159,14 @@ def _l10n_es_edi_facturae_get_corrective_data(self):
             }
         return {}
 
+    def _l10n_es_edi_facturae_get_receiver_transaction_reference(self, line):
+        if (
+                self.env['ir.module.module'].search([('name', '=', 'sale_management')]).state == 'installed'
```
Comment (epictete) [inline addons/l10n_es_edi_facturae/models/account_move.py:164]:
```suggestion
                self.env['ir.module.module']._get('sale_management').state == 'installed'
```

File: addons/l10n_es_edi_facturae/models/account_move.py [hunk 2]
```diff
@@ -252,6 +260,7 @@ def _l10n_es_edi_facturae_inv_lines_to_items(self, conversion_rate=None):
             totals['total_taxes_withheld'] += sum((abs(tax["tax_amount"]) for tax in taxes_withheld_computed))
 
             invoice_line_values.update({
+                'ReceiverTransactionReference': self._l10n_es_edi_facturae_get_receiver_transaction_reference(line),
```
Comment (epictete) [inline addons/l10n_es_edi_facturae/models/account_move.py:263]: Not sure having a dedicated method for this is necessary :thinking: 
Couldn't we simply use a variable?

Comment (alialfie) [reply addons/l10n_es_edi_facturae/models/account_move.py:263]: I did this initially then decided to go with a method instead. Felt it was better not to add more lines to an already  big method but I can revert it if you'd like!

File: addons/l10n_es_edi_facturae/models/account_move.py (hunk 1 above)
Comment (epictete) [reply addons/l10n_es_edi_facturae/models/account_move.py:164]: Even better :+1:

File: addons/l10n_es_edi_facturae/models/account_move.py (hunk 2 above)
Comment (epictete) [reply addons/l10n_es_edi_facturae/models/account_move.py:263]: I think having a method is useful if it needs to be overridden or if the same logic is repeated several times, but in this case, I think a simple variable would be enough :blush:

**Comment by alialfie** [comment]
@epictete thanks for the review! updated :)

**Review by epictete — APPROVED** [review]
Thanks for the quick update :+1: 

@robodoo r+

**Pull request closed without merging**